components/supervisor/pkg/serverapi/publicapi.go

```diff
@@ -375,3 +383,32 @@ func (s *Service) RegisterMetrics(registry *prometheus.Registry) error {
 	}
 	return registry.Register(s.publicApiMetrics)
 }
+
+func workspaceStatusToWorkspaceInstance(status *v1.WorkspaceStatus) *gitpod.WorkspaceInstance {
+	instance := &gitpod.WorkspaceInstance{
+		CreationTime: status.Instance.CreatedAt.String(),
+		ID:           status.Instance.InstanceId,
+		Status: &gitpod.WorkspaceInstanceStatus{
+			ExposedPorts: []*gitpod.WorkspaceInstancePort{},
+			Message:      status.Instance.Status.Message,
+			// OwnerToken:   "", not used so ignore
+			Phase:   status.Instance.Status.Phase.String(),
+			Timeout: status.Instance.Status.Conditions.Timeout,
+			Version: int(status.Instance.Status.StatusVersion),
+		},
+		WorkspaceID: status.Instance.WorkspaceId,
+	}
+	for _, port := range status.Instance.Status.Ports {
+		info := &gitpod.WorkspaceInstancePort{
+			Port: float64(port.Port),
+			URL:  port.Url,
+		}
+		if port.Policy == v1.PortPolicy_PORT_POLICY_PUBLIC {
+			info.Visibility = gitpod.PortVisibilityPublic
+		} else {
+			info.Visibility = gitpod.PortVisibilityPrivate
+		}
```

Comment:
PortPolicy is one of 3 values:
```
	PortPolicy_PORT_POLICY_UNSPECIFIED PortPolicy = 0
	// Private means the port is accessible by the workspace owner only using the workspace port URL
	PortPolicy_PORT_POLICY_PRIVATE PortPolicy = 1
	// Public means the port is accessible by everybody using the workspace port URL
	PortPolicy_PORT_POLICY_PUBLIC PortPolicy = 2
```
Use a switch here and handle them all. This ensures that you actually detect cases where the port policy is not set (UNSPECIFIED)

Reply:
It's the same as here, it can only be public or private and if for some reason is unspecified  it should convert to private either way :thinking: 
https://github.com/gitpod-io/gitpod/blob/6245c323a5cfba200e77fedb0e01b424ed6655cc/components/supervisor/pkg/serverapi/publicapi.go#L406-L410